tests/unit/test_security.py

```diff
@@ -0,0 +1,752 @@
+        x_token_info_func="tests.fixtures.handlers.token_info",
+    )
+    check = build_oauth2_security_check(requirement, scheme)
+    assert callable(check)
+
+
+def test_build_security_check_list():
+    requirements = [
+        ("basic", []),
+        ("bearer", []),
+        ("api_key", []),
+        ("oauth2", ["a"]),
+    ]
+    schemes = dict(
+        basic=SecurityScheme(
+            SecuritySchemesType.HTTP,
+            scheme=HTTPAuthenticationScheme.BASIC,
+            x_basic_info_func="tests.fixtures.handlers.basic_info",
+        ),
+        bearer=SecurityScheme(
+            SecuritySchemesType.HTTP,
+            scheme=HTTPAuthenticationScheme.BEARER,
+            x_bearer_info_func="tests.fixtures.handlers.bearer_info",
+        ),
+        api_key=SecurityScheme(
+            SecuritySchemesType.HTTP_API_KEY,
+            name="api_key",
+            in_=ApiKeyLocation.QUERY,
+            x_api_key_info_func="tests.fixtures.handlers.api_key_info",
+        ),
+        oauth2=SecurityScheme(
+            SecuritySchemesType.OAUTH2,
+            flows=OAuth2Flows(
+                implicit=OAuth2Flow(authorization_url="", scopes={"a": "a"})
+            ),
+            x_token_info_func="tests.fixtures.handlers.token_info",
+        ),
+    )
+
+    check = build_security_handler(requirements, schemes)
+    assert check
+    assert callable(check)
+
+
+def test_build_security_handler_with_invalid_handler():
+    requirements = [{"basic": []}]
+    schemes = dict(
+        basic=SecurityScheme(
+            SecuritySchemesType.HTTP,
+            scheme=HTTPAuthenticationScheme.BASIC,
+            x_basic_info_func="tests.fixtures.handlers.basic_info",
+        )
+    )
+    factory = security_handler_factory(requirements, schemes)
+    with pytest.raises(SecurityException):
+        factory(None)
+
+
+def test_http_basic_works():
+    requirements = [{"basic": []}]
+    schemes = dict(
+        basic=SecurityScheme(
+            SecuritySchemesType.HTTP,
+            scheme=HTTPAuthenticationScheme.BASIC,
+            x_basic_info_func="tests.fixtures.handlers.basic_info",
+        )
+    )
+
+    def on_connect(*args, **kwargs):
+        print(args, kwargs)
+
+    factory = security_handler_factory(requirements, schemes)
+    handler_with_security = factory(on_connect)
+    with Flask(__name__).test_client() as c:
+        basic_auth = base64.b64encode("username:password".encode()).decode()
+        headers = {"Authorization": f"basic {basic_auth}"}
+        c.post(headers=headers)
+        handler_with_security()
+
+
+def test_http_basic_fails():
+    requirements = [{"basic": []}]
+    schemes = dict(
+        basic=SecurityScheme(
+            SecuritySchemesType.HTTP,
+            scheme=HTTPAuthenticationScheme.BASIC,
+            x_basic_info_func="tests.fixtures.handlers.basic_info",
+        )
+    )
+
+    def on_connect(*args, **kwargs):
+        print(args, kwargs)
```

Comment:
Not a great fan of printing stuff within tests

```suggestion
    mock_ack = Mock()
    def on_connect(*args, **kwargs):
        mock_ack()
        
     # and then at the end of the test we can assert whether the actual handler has been called or not
    mock_ack.assert_called_once()
    # or
    mock_ack.assert_not_called()
```

Reply:
The same comment applies for to all other tests that print within the connection handler.